crates/iroha_tree/src/changeset.rs

```diff
@@ -0,0 +1,829 @@
+                    )]
+                    .into_iter()
+                    .collect(),
+                    GrantBox::RolePermission(inst) => [node!(
+                        RolePermission,
+                        inst.destination,
+                        tr::PermissionId::from(&inst.object),
+                        UnitW::Create(())
+                    )]
+                    .into_iter()
+                    .collect(),
+                },
+                InstructionBox::Revoke(inst) => match inst {
+                    RevokeBox::Permission(inst) => [node!(
+                        AccountPermission,
+                        inst.destination.signatory,
+                        inst.destination.domain,
+                        tr::PermissionId::from(&inst.object),
+                        UnitW::Delete(())
+                    )]
+                    .into_iter()
+                    .collect(),
+                    RevokeBox::Role(inst) => [node!(
+                        AccountRole,
+                        inst.destination.signatory,
+                        inst.destination.domain,
+                        inst.object,
+                        UnitW::Delete(())
+                    )]
+                    .into_iter()
+                    .collect(),
+                    RevokeBox::RolePermission(inst) => [node!(
+                        RolePermission,
+                        inst.destination,
+                        tr::PermissionId::from(&inst.object),
+                        UnitW::Delete(())
+                    )]
+                    .into_iter()
+                    .collect(),
+                },
+                InstructionBox::ExecuteTrigger(_inst) => unimplemented!(
+                    "planned to be replaced with calls to pre-registered executables (#5147)"
+                ),
+                InstructionBox::SetParameter(inst) => [node!(
+                    Parameter,
+                    tr::ParameterId,
+                    ParameterW::Set(inst.0.into())
+                )]
+                .into_iter()
+                .collect(),
+                InstructionBox::Upgrade(_inst) => {
+                    [node!(Authorizer, AuthorizerW::Set(state::tr::AuthorizerV))]
+                        .into_iter()
+                        .collect()
+                }
+                InstructionBox::Log(inst) => {
+                    const TARGET: &str = "log_isi";
+                    match inst.level {
+                        dm::Level::TRACE => iroha_logger::trace!(target: TARGET, "{}", inst.msg),
+                        dm::Level::DEBUG => iroha_logger::debug!(target: TARGET, "{}", inst.msg),
+                        dm::Level::INFO => iroha_logger::info!(target: TARGET, "{}", inst.msg),
+                        dm::Level::WARN => iroha_logger::warn!(target: TARGET, "{}", inst.msg),
+                        dm::Level::ERROR => iroha_logger::error!(target: TARGET, "{}", inst.msg),
+                    }
```

Comment:
Side effect should probably not occur during the `TryFrom` conversion, which in all other ways is a pure data transformation.

Isn't critical though, but maybe it's possible to have some sort of "side-effect" node?